index.d.ts

```diff
@@ -0,0 +1,392 @@
+/**
+  The login function will login
+
+  @param {object} [options] - Options object for the deploy function
+  @param {string} [options.projectLocation] - the location(directory) of your projects package.json. Defaults to `process.cwd`
+  @param {string} [options.token] - auth token to pass into the openshift rest client for logging in with the API Server.  Overrides the username/password
+  @param {string} [options.username] - username to pass into the openshift rest client for logging in with the API Server
+  @param {string} [options.password] - password to pass into the openshift rest client for logging in with the API Server
+  @param {string} [options.apiServer] - use server instead. apiServer to pass into the openshift rest client for logging in with the API Server
+  @param {string} [options.server] - server to pass into the openshift rest client for logging in with the API Server
+  @param {string} [options.insecure] - flag to pass into the openshift rest client for logging in with a self signed cert.  Only used with apiServer login.  default to false
+  @param {string} [options.forceLogin] - Force a login when using the apiServer login.  Only used with apiServer login.  default to false
+  @returns {Promise<object>} - Returns a JSON Object
+*/
+export function login(options?: {
+    projectLocation?: string;
+    token?: string;
+    username?: string;
+    password?: string;
+    apiServer?: string;
+    server?: string;
+    insecure?: string;
+    forceLogin?: string;
+}): Promise<object>;
+/**
+  The login function will login
+
+  @param {object} [options] - Options object for the deploy function
```

Comment:
Should this be options for the `logout` function instead of for the `deploy` function?